daphne_worker/src/dap.rs

```diff
@@ -546,20 +547,20 @@ impl<D> DapHelper<BearerToken> for DaphneWorkerConfig<D> {
             }
         }
 
-        let mut futures = Vec::new();
+        let mut requests = Vec::new();
         for (durable_name, nonce_hex_set) in durable_requests.into_iter() {
-            futures.push(self.durable_post(
+            requests.push(self.durable_post(
                 BINDING_DAP_REPORT_STORE,
                 DURABLE_REPORT_STORE_PUT_PROCESSED,
                 durable_name,
                 nonce_hex_set,
             ));
         }
 
+        let responses: Vec<Vec<(String, TransitionFailure)>> = try_join_all(requests).await?;
```

Comment:
Is this what we want? If any future fails then all futures are cancelled.

Reply:
Great question. Any failure here would be a failure of the Workers runtime itself and would be handled by DaphneWorker as a 500 internal error.

It might be possible to recover by, say, "unmarking" any reports as aggregated. But this would require another call to DO instances, which is failing under this scenario :(

Thus I think hard fail is appropriate here.